src/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -748,20 +753,70 @@ const UsersConfigurationRow = ({
+
+    const handleClear = (_event) => {
+        onChange(idx, "file", "");
+        onChange(idx, "value", "");
+    };
+
+    return (
+        <Grid id={id}>
+            <GridItem span={11}>
+                <FileUpload id={"create-vm-dialog-ssh-key-" + id}
+                    type="text"
+                    value={item.value}
+                    filename={item.filename}
+                    filenamePlaceholder="Drag and drop a file or upload one"
+                    onFileInputChange={(_event, file) => onChange(idx, "file", file)}
+                    onDataChange={(_event, value) => onChange(idx, "value", value)}
+                    onTextChange={(_event, value) => onChange(idx, "value", value)}
+                    onReadStarted={() => setIsLoading(true)}
+                    onReadFinished={() => setIsLoading(false)}
+                    onClearClick={_event => handleClear(_event)}
+                    isLoading={isLoading}
+                    allowEditingUploadedText
+                    browseButtonText="Upload" />
+            </GridItem>
+            <GridItem span={1} className="pf-m-1-col-on-md remove-button-group">
+                <Button variant='plain'
+                    className="btn-close"
+                    id={id + "-btn-close"}
+                    isSmall
+                    aria-label={_("Remove item")}
+                    icon={<TrashIcon />}
+                    onClick={() => removeitem(idx)} />
+            </GridItem>
+        </Grid>
+    );
+};
+
 const CloudInitOptionsRow = ({
     onValueChanged,
     rootPassword,
     userLogin, userPassword,
     validationFailed,
 }) => {
     return (
-        <UsersConfigurationRow rootPassword={rootPassword}
-                               rootPasswordLabelInfo={_("Leave the password blank if you do not wish to set a root password")}
-                               showUserFields
-                               userLogin={userLogin}
-                               userPassword={userPassword}
-                               validationFailed={validationFailed}
-                               onValueChanged={onValueChanged} />
+        <>
+            <UsersConfigurationRow rootPassword={rootPassword}
+                                   rootPasswordLabelInfo={_("Leave the password blank if you do not wish to set a root password")}
+                                   showUserFields
+                                   userLogin={userLogin}
+                                   userPassword={userPassword}
+                                   validationFailed={validationFailed}
+                                   onValueChanged={onValueChanged} />
+            <DynamicListForm id="create-vm-dialog-ssh-key"
+                emptyStateString={_("No SSH keys specified")}
+                label={_("SSH keys")}
+                actionLabel={_("Add SSH keys")}
+                onChange={value => onValueChanged('sshKeys', value)}
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1150-20230814-091115-8af58863-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)